packages/salesforcedx-vscode-core/src/commands/forceAuthWebLogin.ts

```diff
@@ -87,6 +83,42 @@ export class ForceAuthWebDemoModeLoginExecutor extends SfdxCommandletExecutor<{}
   }
 }
 
+export class ForceAuthWebLoginDemoModeExecutor extends ForceAuthDemoModeExecutor<Alias> {
+  public build(data: Alias): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('force_auth_web_login_authorize_org_text')
+      )
+      .withArg('force:auth:web:login')
+      .withFlag('--setalias', data.alias)
+      .withArg('--setdefaultusername')
+      .withArg('--noprompt')
+      .withJson()
+      .build();
+  }
+}
+
+export class AliasGatherer implements ParametersGatherer<Alias> {
```

Comment:
This code was copied from forceOrgCreate.ts so that I could customize the default alias. Is there a better way for me to reuse the AliasGatherer from that file while still being able to customize the default alias?

Reply:
I think this is fine. The alternative would be to subclass (not a fan of subclassing) or to create a new method that can be delegated to.